openslides/utils/consumers.py

```diff
@@ -136,6 +137,19 @@ class SiteConsumer(ProtocollAsyncJsonWebsocketConsumer):
                 )
             else:
                 await self.send_json(type='error', content='Invalid notify message', in_response=id)
+        elif type == 'constantsRequest':
+            angular_constants: Dict[str, Any] = {}
+            for app in apps.get_app_configs():
+                try:
+                    # Each app can deliver values to angular when implementing this method.
+                    # It should return a list with dicts containing the 'name' and 'value'.
+                    get_angular_constants = app.get_angular_constants
+                except AttributeError:
+                    # The app doesn't have this method. Continue to next app.
+                    continue
+                constants = await database_sync_to_async(get_angular_constants)()
```

Comment:
I want the SiteConsumer to be fully async without any switch to sync (which starts new threads). Specially database requests in another thread are problematic, see: https://github.com/django/channels/issues/1110  

the angular_constants should be real constants. Therefore I would suggest, that you load all constants into a global dict on startup. Then you only have to return this dict here.

To make it even clearer, that the constants are constants, I would say, we remove the config vars from the constants. They are problematic. For example when you render the index.html the constants are hard coded into the rendered file. When you change the config vars afterwards, the change has no effect, which can be confusing.

Therefore "variable" constants should be in the settings.py to make in clearer, that they can not be changed at runtime.